File: GitToolBox/src/main/java/zielu/gittoolbox/fetch/AutoFetchNotifier.java
package zielu.gittoolbox.fetch;

import com.intellij.util.messages.Topic;
import org.jetbrains.annotations.NotNull;

public interface AutoFetchNotifier {
  Topic<AutoFetchNotifier> TOPIC = Topic.create("Git ToolBox Auto Fetch", AutoFetchNotifier.class);

  void stateChanged(@NotNull AutoFetchState state);
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/fetch/AutoFetchNotifier.kt
package zielu.gittoolbox.fetch

import com.intellij.util.messages.Topic

internal interface AutoFetchNotifier {
  fun stateChanged(state: AutoFetchState)

  companion object {
    val TOPIC = Topic.create("Git ToolBox Auto Fetch", AutoFetchNotifier::class.java)
  }
}
